api/src/tests/api.test.ts

```diff
@@ -957,4 +957,81 @@ describe("API", () => {
       expect(await knex("entitydata").select()).toHaveLength(0);
     });
   });
+
+  describe("GET /api/v0/papers/arxiv:{arxivId}/version", () => {
+    test("paper with entity data", async () => {
+      await knex("paper").insert({
+        s2_id: "s2id",
+        arxiv_id: "1111.1111v1",
+      } as PaperRow);
+      await knex("version").insert({
+        id: 1,
+        paper_id: "s2id",
+        index: 0,
+      } as VersionRow);
+      await knex.batchInsert("entity", [
+        {
+          id: 1,
+          paper_id: "s2id",
+          version: 0,
+          type: "unknown",
+          source: "test",
+        },
+      ] as EntityRow[]);
+      await knex.batchInsert("boundingbox", [
+        {
+          entity_id: 1,
+          source: "test",
+          page: 0,
+          left: 0,
+          top: 0,
+          width: 0.1,
+          height: 0.05,
+        },
+      ] as BoundingBoxRow[]);
+      await knex.batchInsert("entitydata", [
```

Comment:
I note that the inserts into `boundingbox` and `entitydata` are not strictly necessary to pass the test. Is it worth removing them for the sake of test case simplicity? Or perhaps only passing the version check if entities have bounding boxes / data?

Reply:
I'll give that a try. It's hard to tell what setup is necessary with these heavyweight integration tests that require actual data in the DB to test API behaviors. Ideally our tests would allow me to mock the DB connection and write unit tests on just the API layer instead.

Reply:
Fair enough. I would be interested in discussing how make the tests less heavy-handed. Admittedly I did want to have tests for the SQL queries since some of them get pretty complex. That said, it is annoying that the tests require so much setup.